src/pages/home/report/ReportActionListFrozenScrollContext.js

```diff
@@ -0,0 +1,67 @@
+import React, {createContext, forwardRef, useMemo, useState} from "react";
+import PropTypes from "prop-types";
+import getComponentDisplayName from "../../../libs/getComponentDisplayName";
+
+
+const withScrollFrozenPropTypes = {
+    /** flag determining if we should freeze the scroll */
+    shouldFreezeScroll: PropTypes.bool,
+
+    /** Function to update the state */
+    setShouldFreezeScroll: PropTypes.func,
+};
+
+const ReportActionListFrozenScrollContext = createContext(null);
+
+
+function ReportActionListFrozenScrollContextProvider(props) {
+    const [shouldFreezeScroll, setShouldFreezeScroll] = useState(false);
+
+    /**
+     * The context this component exposes to child components
+     * @returns {Object} flag and a flag setter
+     */
+    const contextValue = useMemo(
+        () => ({
+            shouldFreezeScroll,
+            setShouldFreezeScroll,
+        }),
+        [shouldFreezeScroll, setShouldFreezeScroll],
+    );
+
+    return <ReportActionListFrozenScrollContext.Provider
+        value={contextValue}>{props.children}</ReportActionListFrozenScrollContext.Provider>;
```

Comment:
The `useMemo` looks like an overhead here.
cc @fabioh8010 @gedu 
```suggestion
        value={{
            shouldFreezeScroll,
            setShouldFreezeScroll,
        }}>{props.children}</ReportActionListFrozenScrollContext.Provider>;
```

Reply:
I think it's correct @rezkiy37, because if we pass an object in this way, is like we are creating a new object every time and it will trigger unnecessary re-renders (old object will never equal newly created one). It makes sense to me to memoize.

Reply:
This does not need to be memorized. We could return the `useState`object to the provider (as an array, not object), which will be the same reference. WDYT? 
```
function ReportActionListFrozenScrollContextProvider(props) {
    const shouldFreezeScrollState = useState(false);
    return (
        <ReportActionListFrozenScrollContext.Provider value={shouldFreezeScrollState}>
            {props.children}
        </ReportActionListFrozenScrollContext.Provider>
    );
}
```

Reply:
Agree with @ArekChr approach.